### PR TITLE
Unlock operating limits cheat - both upper and lower bound checking

### PR DESCRIPTION
**Pull request by octaroot**
Fixes the hill-lift portion of issue #3492, better checks need to be in place in order to properly check both bounds when not using the `gCheatsFastLiftHill` cheat. The problem is that some operational modes (such as launch speed) don't have separate methods and are handled via generic `window_ride_mode_tweak_increase` (or `decrease`) while different bounds apply to different operational modes.

### CODE REVIEW

**Comment by octaroot** [comment]
I've revisited my commit and refactored for better readability. I also added bound checks for all other operation modes. This should fix all problems related to:

1. turning on cheat
2. setting the operation mode out of vanilla-limits (e.g. high hill lift speed)
3. turning the cheat off again
4. attempting to adjust the now out-of-bounds operating mode (either increasing or decreasing would cause an error and not change at all).

**I've created a simple macro. Is this acceptable?**

Thank you for reviewing the code :-)

File: src/windows/ride.c [hunk 1]
```diff
@@ -39,6 +39,7 @@
 #include "../rct1.h"
 
 #define var_496(w)	RCT2_GLOBAL((int)w + 0x496, uint16)
+#define clamp_value(value, min_value, max_value) max(min(max_value, value), min_value)
```
Comment (IntelOrca) [inline src/windows/ride.c]: Already defined in rct2.h

Comment (octaroot) [inline src/windows/ride.c]: Whoops, didn't notice that one. Refactored.

**Comment by duncanspumpkin** [comment]
Have you checked that there are no issues with overflows and underflows?

**Comment by octaroot** [comment]
I checked a couple of rides, namely the ones with separate checks (hill-lift, circuits), generic wait times and rides with launch speed. All worked fine, otherwise I wouldn't have opened a PR. I didn't check _every_ ride there is. Also the code didn't introduce new bounds - it just clamps to existing ones.

**Pull request merged**